### PR TITLE
[CBRD-23140] set head_lsa as tail_lsa for first log record of transaction

### PR DESCRIPTION
**Pull request by alexstanro**
http://jira.cubrid.org/browse/CBRD-23140

### CODE REVIEW

**Comment by eseokoh** [comment]
@alexstanro 
What is a case that head_lsa is null and tail_lsa is not?

**Comment by alexstanro** [comment]
> @alexstanro
> What is a case that head_lsa is null and tail_lsa is not?

Please, notice prior_lsa_start_append. If the transaction modify only heap (there is no index), there will be only one log record and NULL head_lsa, NOT NULL tail_lsa.

**Comment by ribram** [comment]
> > @alexstanro
> > What is a case that head_lsa is null and tail_lsa is not?
> 
> Please, notice prior_lsa_start_append. If the transaction modify only heap (there is no index), there will be only one log record and NULL head_lsa, NOT NULL tail_lsa.

I remembered changing the code and it looks like a regression of that change.

Please see e1b88d38f6 #1643. tail_lsa used to be set before copying to head_lsa.

File: src/transaction/log_append.cpp [hunk 1]
```diff
@@ -1560,16 +1560,23 @@ prior_lsa_start_append (THREAD_ENTRY *thread_p, LOG_PRIOR_NODE *node, LOG_TDES *
...
       LSA_COPY (&node->log_header.prev_tranlsa, &tdes->tail_lsa);
 
+      LSA_COPY (&tdes->tail_lsa, &log_Gl.prior_info.prior_lsa);
+
       /*
        * Is this the first log record of transaction ?
        */
       if (LSA_ISNULL (&tdes->head_lsa))
 	{
+          /* We may have NULL head LSA and not NULL tail LSA, if the transaction
```
Comment (eseokoh) [inline src/transaction/log_append.cpp]: I think you can delete this comment, since now head == tail.

**Pull request merged**